test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresInCompilerGeneratedCode.cs

```diff
@@ -314,13 +387,25 @@ static async IAsyncEnumerable<int> TestReflectionAccess ()
 			}
 
 			[ExpectedWarning ("IL2026", "--MethodWithRequires--", CompilerGeneratedCode = true)]
+			[ExpectedWarning ("IL3002", "--MethodWithRequires--", ProducedBy = ProducedBy.Analyzer)]
 			static async IAsyncEnumerable<int> TestLdftn ()
 			{
 				await MethodAsync ();
 				yield return 0;
 				var action = new Action (MethodWithRequires);
 			}
 
+			[ExpectedWarning ("IL2026", "Message from --InitString--", CompilerGeneratedCode = true)]
```

Comment:
I don't think these should have the `CompilerGeneratedCode = true` on them... that is meant for cases where the origin is from the compiler generated code (and thus it's not possible to put the attribute there). In this case the warning happens on a static field directly - so no need for this.

Reply:
According to linker the warning comes from the static constructor, for example:
`ILLink: Trim analysis warning IL2026: Mono.Linker.Tests.Cases.RequiresCapability.RequiresInCompilerGeneratedCode.SuppressInAsyncBody..cctor(): Using member 'Mono.Linker.Tests.Cases.RequiresCapability.RequiresInCompilerGeneratedCode.InitString()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Message from --InitString--.` so the CompilerGeneratedCode is needed. In fact I think I forgot to add some code handling that I had before

Reply:
OK _ I guess we can treat static .ctors as compiler generated code. This property was initially designed for the async/iterator state machine code, but it seems to make sense in other cases as well 😄

Reply:
Added an additional piece of code in the result checker to verify the static constructor call (was already in https://github.com/dotnet/linker/pull/2113)